app/views/play/level/ControlBarView.coffee

```diff
@@ -38,13 +43,37 @@ module.exports = class ControlBarView extends CocoView
     @levelID = @levelSlug or @level.id
     @spectateGame = options.spectateGame ? false
     @observing = options.session.get('creator') isnt me.id
+
+    @levelNumber = ''
+    if @level.get('type') is 'course' and @level.get('campaignIndex')?
+      @levelNumber = @level.get('campaignIndex') + 1
+    if @courseInstanceID
+      @courseInstance = new CourseInstance(_id: @courseInstanceID)
+      @supermodel.trackRequest(@courseInstance.fetch().then(=>
+        @classroom = new Classroom(_id: @courseInstance.get('classroomID'))
+        @supermodel.trackRequest @classroom.fetch()
+      ))
```

Comment:
Ok.  Seemed to work when I tried it, but I'll update the calls to use your suggested method.